src/py_obs/osc.py

```diff
@@ -49,14 +53,41 @@ async def api_request(
         except aiohttp.ClientResponseError as cre_exc:
             raise ObsException(**cre_exc.__dict__) from cre_exc
 
+    async def get_auth_header(self) -> hash:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(self.ibs_url_about) as r:
+                challenge_parameters = {
+                    'realm': r.headers['WWW-Authenticate'].split(',')[0].split('=')[-1].replace('"',""),
+                    'payload': r.headers['WWW-Authenticate'].split(',')[1].split('=')[-1].replace('"',"")
+                }
+                CREATED_TIMESTAMP = str(int(time.time()))
+                SIGNATURE_STRING = f"(created): {CREATED_TIMESTAMP}"
+                cmd = ['ssh-keygen', '-Y', 'sign', '-f', self.private_key_path, '-q', '-n',
+                        challenge_parameters['realm']]
+                s = subprocess.run(cmd, input=SIGNATURE_STRING, text=True, capture_output=True)
+
+                signature = ''.join(s.stdout.split('\n')[1:-2])
+                auth_header = { 'Authorization': f'Signature keyId="{self.username}",algorithm="ssh", signature="{signature}",headers="(created)",created="{CREATED_TIMESTAMP}"'}
+            async with session.get(self.ibs_url_about, headers=auth_header) as r:
+                assert str(r.status) == '200'
```

Comment:
Python will by default skip assertions in production, so don't rely on assertions. Use `raise_for_status` instead.
```suggestion
            async with session.get(self.ibs_url_about, headers=auth_header, raise_for_status=True) as r:
```

Reply:
OK done